File: python3/LeetCode/singleNumber.py
import collections
from typing import List

class Solution:
    def singleNumber(self, nums:List[int]) ->int:
        frep = collections.Counter(nums)
        res = [num for num, occ in frep.items() if occ == 1][0]
        return res

    def singleNumber1(self, nums:List[int]) ->int:
        res = 0
        for i in range(0,32):
            total = sum((num >> i)&1 for num in nums)
            if total % 3:
                if i==31:
                    res -=(1<<i)
                else:
                    res |=(1<<i)
        return res


if __name__ == '__main__':
    nums = [1,1,1,3]
    print(Solution().singleNumber1(nums))

File: go/LeetCode/singleNumber.go
package main

import "fmt"

func main() {
	nums := []int{1,1,1,3}
	fmt.Print(singleNumber1(nums))
}

func singleNumber(nums []int) int {
	count := make(map[int]int)

	for _,k :=range nums{
		count[k]++
	}

	res := 0
	for k,v :=range count{
		if v == 1 {
			res = k
			break
		}
	}
	return res
}

func singleNumber1(nums []int) int32 {
	res := int32(0)
	for i:=0;i<32;i++{
		total := int32(0)
		for _,num := range nums{
			total += (int32(num) >> i) &1
		}
		if total%3 == 1 {
			res |= 1<<i
		}
	}
	return res
}
